src/rpc/client.rs

```diff
@@ -92,38 +93,33 @@ impl Client {
         }
     }
 
-    pub fn report_response(&mut self) -> Result<(), Error> {
+    pub fn report_response_or_fail(&mut self) -> Result<(), Error> {
         let resp = self.report_failure()?;
         // note: this triggers the yaml formatting when implemented
-        println!("{:#}", resp);
+        println!("{}", resp);
         Ok(())
     }
 
-    pub fn report_progress(&mut self) -> Result<usize, Error> {
-        let mut counter = 0;
-        let mut finished = false;
-        while !finished {
-            finished = true;
-            counter += 1;
-            match self.report_failure()? {
-                // Failure is already covered by `report_response()`
-                Request::Progress(info) => {
-                    println!("{}", info.green_bold());
-                    finished = false;
+    /// Print the stream of received requests until progress fails or succeed
+    pub fn report_progress(&mut self) -> Result<(), Error> {
+        // loop on all requests received until a progress terminaison condition is recieved
```

Comment:
typo s/terminaison/termination/

Reply:
Done in be8d233d9545c3143bdc3030b7587b21cade6952